components/earnings/commons/earnings_base.lua

```diff
@@ -0,0 +1,185 @@
+	conditions = Earnings._buildConditions(conditions, year, mode, startYear)
+
+	if String.isNotEmpty(startYear) then
+		return Earnings._calculatePerYear(conditions, divisionFactor)
+	end
+
+	local lpdbQueryData = mw.ext.LiquipediaDB.lpdb('placement', {
+		conditions = conditions,
+		query = 'sum::prizemoney, mode',
+		groupby = 'mode asc'
+	})
+
+	local totalEarnings = 0
+
+	for _, item in ipairs(lpdbQueryData) do
+		if item['sum_prizemoney'] ~= nil then
+			local prizeMoney = item['sum_prizemoney']
+			totalEarnings = totalEarnings + (prizeMoney / divisionFactor(item['mode']))
+		end
+	end
+
+	return MathUtils._round(totalEarnings, 2)
+end
+
+function Earnings._calculatePerYear(conditions, divisionFactor)
+	local totalEarningsOfYear = {}
+	local totalEarnings = {}
+	totalEarnings.total = 0
+
+	local offset = 0
+	local count = 5000
+	while count == 5000 do
+		local lpdbQueryData = mw.ext.LiquipediaDB.lpdb('placement', {
+			conditions = conditions,
+			query = 'prizemoney, mode, date',
+			limit = 5000,
+			offset = offset
+		})
+		for _, item in pairs(lpdbQueryData) do
+			local prizeMoney = tonumber(item.prizemoney) or 0
+			local year = string.sub(item.date, 1, 4)
+			prizeMoney = prizeMoney / divisionFactor(item['mode'])
+			totalEarningsOfYear[year] = (totalEarningsOfYear[year] or 0) + prizeMoney
+		end
+		count = #lpdbQueryData
+		offset = offset + 5000
+	end
+
+	for year, earningsOfYear in pairs(totalEarningsOfYear) do
+		totalEarnings[tonumber(year)] = MathUtils._round(earningsOfYear, 2)
+		totalEarnings.total = totalEarnings.total + earningsOfYear
+	end
+	totalEarnings.total = MathUtils._round(totalEarnings.total, 2)
+
+	return totalEarnings
+end
+
+function Earnings._buildConditions(conditions, year, mode, startYear)
+	conditions = '[[date::!' .. _DEFAULT_DATE .. ']] AND [[prizemoney::>0]] AND ' .. conditions
+	if String.isNotEmpty(startYear) then
+		conditions = conditions .. ' AND (' ..
+			'[[date::>' .. startYear .. _FIRST_DAY_OF_YEAR .. ']] ' ..
+			'OR [[date::' .. startYear .. _FIRST_DAY_OF_YEAR .. ']])'
+	elseif String.isNotEmpty(year) then
+		conditions = conditions .. ' AND (' ..
+			'[[date::>' .. year .. _FIRST_DAY_OF_YEAR .. ']] ' ..
+			'OR [[date::' .. year .. _FIRST_DAY_OF_YEAR .. ']]' ..
+			') AND (' ..
+			'[[date::<' .. year .. _LAST_DAY_OF_YEAR .. ']] ' ..
+			'OR [[date::' .. year .. _LAST_DAY_OF_YEAR .. ']])'
+	end
+
+	if String.isNotEmpty(mode) then
+		conditions = conditions .. ' AND [[mode::' .. mode .. ']]'
+	end
+
+	return conditions
+end
+
+-- overwritable in /Custom
+local _DEFAULT_NUMBER_OF_PLAYERS_IN_TEAM = 5
+
+-- overwritable in /Custom
+function Earnings.divisionFactor(mode)
+	if mode == '4v4' then
+		return 4
+	elseif mode == '3v3' then
+		return 3
+	elseif mode == '2v2' then
+		return 2
+	elseif mode == '1v1' or mode == 'individual' then
+		return 1
+	end
+
+	return _DEFAULT_NUMBER_OF_PLAYERS_IN_TEAM
+end
+
+function Earnings._divisionFactorOne()
+	return 1
+end
```

Comment:
Might as well call this divisionFactorTeam(), make it public and customizable. Just like the player variant, have the mode as a parameter.